data-prepper-plugins/otel-trace-group-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/oteltracegroup/OTelTraceGroupPrepperTests.java

```diff
@@ -0,0 +1,239 @@
+    private static final String TEST_TRACE_GROUP_2 = "/test_trace_group_2";
+    private static final String TEST_RAW_SPAN_COMPLETE_JSON_FILE_1 = "raw-span-complete-1.json";
+    private static final String TEST_RAW_SPAN_COMPLETE_JSON_FILE_2 = "raw-span-complete-2.json";
+    private static final String TEST_RAW_SPAN_MISSING_TRACE_GROUP_JSON_FILE_1 = "raw-span-missing-trace-group-1.json";
+    private static final String TEST_RAW_SPAN_MISSING_TRACE_GROUP_JSON_FILE_2 = "raw-span-missing-trace-group-2.json";
+    private static final int TEST_NUM_WORKERS = 2;
+
+    private MockedStatic<ConnectionConfiguration> connectionConfigurationMockedStatic;
+
+    private OTelTraceGroupPrepper otelTraceGroupPrepper;
+    private ExecutorService executorService;
+
+    @Mock
+    private ConnectionConfiguration connectionConfigurationMock;
+
+    @Mock
+    private RestHighLevelClient restHighLevelClient;
+
+    @Mock
+    private SearchResponse testSearchResponse;
+
+    @Mock
+    private SearchHits testSearchHits;
+
+    @Mock
+    private SearchHit testSearchHit1;
+
+    @Mock
+    private SearchHit testSearchHit2;
+
+    @Before
+    public void setUp() throws Exception{
+        connectionConfigurationMockedStatic = Mockito.mockStatic(ConnectionConfiguration.class);
+        connectionConfigurationMockedStatic.when(() -> ConnectionConfiguration.readConnectionConfiguration(any(PluginSetting.class)))
+                .thenReturn(connectionConfigurationMock);
+        when(connectionConfigurationMock.createClient()).thenReturn(restHighLevelClient);
+        when(restHighLevelClient.search(any(SearchRequest.class), any(RequestOptions.class))).thenReturn(testSearchResponse);
+        doNothing().when(restHighLevelClient).close();
+        when(testSearchResponse.getHits()).thenReturn(testSearchHits);
+        when(testSearchHits.getHits()).thenReturn(new SearchHit[] {testSearchHit1});
+        when(testSearchHit1.field("traceId")).thenReturn(new DocumentField("traceId", Collections.singletonList(TEST_TRACE_ID_1)));
+        when(testSearchHit1.field("traceGroup")).thenReturn(new DocumentField("traceGroup", Collections.singletonList(TEST_TRACE_GROUP_1)));
+        when(testSearchHit2.field("traceId")).thenReturn(new DocumentField("traceId", Collections.singletonList(TEST_TRACE_ID_2)));
+        when(testSearchHit2.field("traceGroup")).thenReturn(new DocumentField("traceGroup", Collections.singletonList(TEST_TRACE_GROUP_2)));
+        final PluginSetting testPluginSetting = new PluginSetting("otel_trace_group_prepper", new HashMap<>()) {{
+            setPipelineName("testPipelineName");
+        }};
+        otelTraceGroupPrepper = new OTelTraceGroupPrepper(testPluginSetting);
+        executorService = Executors.newFixedThreadPool(TEST_NUM_WORKERS);
+    }
+
+    @After
+    public void tearDown() {
+        otelTraceGroupPrepper.shutdown();
+        connectionConfigurationMockedStatic.close();
```

Comment:
Does removing this cause any harm? I'm not sure how calling a method on a mocked object is necessary (also I can't remember the prior PR where we had issues) :-/

Reply:
Yes. It will interfere with tests where the actual ConnectionConfiguration is being used. Can't remember the prior PR number but it is in peer-forwarder: https://github.com/opendistro-for-elasticsearch/data-prepper/blob/a6b84f22ada1615e5102f1a2c9c7793d5da2acd5/data-prepper-plugins/peer-forwarder/src/test/java/com/amazon/dataprepper/plugins/prepper/peerforwarder/PeerForwarderTest.java#L79